src/util/PermalinkUtil/PermalinkUtil.ts

```diff
@@ -0,0 +1,64 @@
+import OlMap from 'ol/Map';
+import OlLayer from 'ol/layer/Base';
+import TileLayer from 'ol/layer/Tile';
+import MapUtil from '@terrestris/ol-util/dist/MapUtil/MapUtil';
+
+/**
+ * Helper class for some operations related to permalink function.
+ *
+ * @class
+ */
+export class PermalinkUtil {
+
+  /**
+   * Creates a permalink based on the given map state
+   * Permalink will contain center, zoom and layers, using semicolon as
+   * separator for arrays
+   * @param {OlMap} map The openlayers map
+   * @return {string} The permalink.
+   */
+  static getLink = (map: OlMap): string => {
+    const center = map.getView().getCenter().join(';');
+    const zoom = map.getView().getZoom();
+    const layers = MapUtil.getAllLayers(map);
+    const visibles = layers.filter((l: OlLayer) => l instanceof TileLayer &&
+      l.getVisible()).map((l: TileLayer) => l.get('name')).join(';');
+    const link = new URL(window.location.href);
+
+    link.searchParams.set('center', center);
+    link.searchParams.set('zoom', zoom.toString());
+    link.searchParams.set('layers', visibles);
+
+    return link.href;
+  };
```

Comment:
It would be very helpful for project implementations if this logic could be extended for custom parameters easily.

Reply:
well the whole function can be given as custom argument and i think its the better way to go as project needs are often very different